PWG/EMCAL/EMCALbase/AliAnalysisTaskEmcalEmbeddingHelper.cxx

```diff
@@ -615,6 +623,83 @@ bool AliAnalysisTaskEmcalEmbeddingHelper::IsRunInRunlist(const std::string & pat
   return false;
 }
 
+/**
+ * Get the data background run number from the data file path, fRunNumberBgk and 
+ * Determine the MC run block range,  fEmbeddedRunblockMin and  fEmbeddedRunblockMax (used in IsRunInRunblock())
+ * Right now only done once, for the first data file
+ */
+void AliAnalysisTaskEmcalEmbeddingHelper::SetRunblockRange() 
+{
+  Int_t nBlocks = fEmbeddedRunblock.size();
+  
+  if ( nBlocks == 0 ) return ;
+  
+  AliDebug(1,Form("AliAnalysisTaskEmcalEmbeddingHelper::SetRunblockRange() - nRunBlocks %d\n", nBlocks));
+  
+  // Recover the run number from the input data file path.
+  //
+  // InputEvent() seems to be connected to the external event
+  // so one cannot use InputEvent()->GetRunNumber()
+  AliAnalysisManager *mgr = AliAnalysisManager::GetAnalysisManager();
+   
+   if ( !mgr ) return ;
+  
+  Int_t runNumber = mgr->GetRunFromPath(); 
+  if ( fDataRunNumber != runNumber )
+  {
+    //printf("\t Run number, new %d, previous %d\n",runNumber,fDataRunNumber);
+    fDataRunNumber = runNumber;
+  }
+  
+  Int_t  runMin = -1,      runMax = -1;
+  for (Int_t iblock = 0; iblock < nBlocks-1; iblock++) 
+  {
+    std::istringstream(fEmbeddedRunblock.at(iblock  )) >> runMin;
+    std::istringstream(fEmbeddedRunblock.at(iblock+1)) >> runMax;
+    //    runMin = std::stoi(fEmbeddedRunblock.at(iblock  ));
+    //    runMax = std::stoi(fEmbeddedRunblock.at(iblock+1));
+    AliDebug(1,Form("\t block %d, run min %d, run max %d\n",iblock,runMin,runMax));
+    
+    if ( runMin <= fDataRunNumber && runMax > fDataRunNumber ) 
+    {
+      fEmbeddedRunblockMin = runMin;
+      fEmbeddedRunblockMax = runMax;
+      AliDebug(1,Form("\t Selected range %d, run min %d, run max %d\n",
+                      iblock, fEmbeddedRunblockMin, fEmbeddedRunblockMax));
+      
+      return ;
+    }
+  } // block loop
```

Comment:
You may want to checkout `std::lower_bound` and `std::upper_bound` (in conjunction with converting to std::vector). It will do all of this for you. You just have to be a little careful with the index

Reply:
I am not sure how those work, I prefer to stick to what I know, but you can change it